ui/components/multichain/network-list-menu/network-list-menu.tsx

```diff
@@ -305,6 +311,36 @@ export const NetworkListMenu = ({ onClose }: { onClose: () => void }) => {
     const canDeleteNetwork =
       isUnlocked && !isCurrentNetwork && network.chainId !== CHAIN_IDS.MAINNET;
 
+    const onClickCallback = useCallback(() => {
+      handleNetworkChange(network);
+    }, [handleNetworkChange, network]);
+
+    const onDeleteClickCallback = useCallback(() => {
+      dispatch(toggleNetworkMenu());
+      dispatch(
+        showModal({
+          name: 'CONFIRM_DELETE_NETWORK',
+          target: network.chainId,
+          onConfirm: () => undefined,
+        }),
+      );
+    }, [canDeleteNetwork, dispatch, toggleNetworkMenu, showModal, network]);
```

Comment:
You can further refine this callback by only including `network.chainId` instead of the full network.

Reply:
I'll push to get ESLint support for TS React files, that way we can "auto-fix"/auto-complete the dependency arrays